magento/adobe-stock-integration
# [MFTF] Message disappear when the request with broken credentials is sent second time

### Description (*)

This PR will fix the error message issue when user has not configured the ASI configuration. In order to fix this issue, I have extended the `Magento_Ui/js/grid/data-storage` component to cache the custom error message in the masonry data provider
### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#863: Message disappear when the request with broken credentials is sent second time

### Manual testing scenarios (*)

1. User navigates to Stores > Settings > Configuration
2. User expands the `Advanced` tab
3. User clicks on `System` section
4. User expands Adobe Stock Integration expandable section
5. User sets "Enabled Adobe Stock" to "Yes".
6. User saves configuration without providing credentials.
7. Navigate to Cms Pages
8. User clicks Add New Page button
9. Expand the "Content" section
10. Click "Insert Image..." button
11. Click "Search Adobe Stock"
12. Verify that User sees the message
```
Failed to authenticate to Adobe Stock API.
Please correct the API credentials in Configuration → System → Adobe Stock Integration.
```
13. In the pager navigation select 48 instead of 32
Verify that User sees the message
```
Failed to authenticate to Adobe Stock API.
Please correct the API credentials in Configuration → System → Adobe Stock Integration.
```
14. In the pager navigation select 32 instead of 48

### Expected Result
```
Failed to authenticate to Adobe Stock API.
Please correct the API credentials in Configuration → System → Adobe Stock Integration.
```

Depends on: https://github.com/magento/magento2/pull/26502
